pkg/store/proxy_merge.go

```diff
@@ -101,15 +104,15 @@ func (d *responseDeduplicator) Next() bool {
 			continue
 		}
 
-		d.bufferedResp = append(d.bufferedResp, chainSeriesAndRemIdenticalChunks(d.bufferedSameSeries))
+		d.bufferedResp = append(d.bufferedResp, d.chainSeriesAndRemIdenticalChunks(d.bufferedSameSeries))
 		d.prev = s
 
 		return true
 	}
 }
 
-func chainSeriesAndRemIdenticalChunks(series []*storepb.SeriesResponse) *storepb.SeriesResponse {
-	chunkDedupMap := map[uint64]*storepb.AggrChunk{}
+func (d *responseDeduplicator) chainSeriesAndRemIdenticalChunks(series []*storepb.SeriesResponse) *storepb.SeriesResponse {
```

Comment:
One question unrelated to this PR. Is the main usecase of `chainSeriesAndRemIdenticalChunks` to dedup chunks from Receiver? How likely those receiver replicas have exact the same chunk

Reply:
Yeah, it depends on the replication factor. We have RF=2 so there's always at least one extra copy.